### ISSUE TITLE
`JdkObsolete` and Guava's `SortedSetMultimap`

### ISSUE THREAD

**Issue by Stephan202** [google/error-prone#1610]
As of #1594 `JdkObsolete` flags usages of `SortedSet`. However, Guava's `SortedSetMultimap` references this type. Code such as the following now triggers a warning:
```java
SortedSetMultimap<String, String> myMultimap = TreeMultimap.create();
...
String myValue = myMultimap.get("foo");
```

The same applies e.g. to code that uses the result of `Multimaps.asMap(myMultimap)`, even if the resultant map is iterated over using code that does not explicitly reference `SortedSet`.

Is this intentional? If so: should I file an issue with the Guava team to replace all usages of `SortedSet`? (Not sure they'd accept it, though: the required code changes would not be binary backwards compatible.)

As an alternative, perhaps the check should be updated to flag only places where `SortedSet` is (a) explicitly referenced and (b) can be replaced with `NavigableSet` without breaking the code. (The second criterion may include refraining from changing `SortedSet` parameters on non-private methods; accepting a `SortedSet` instead of a `NavigableSet` may be more user-friendly, after all.) These suggestions also apply to `SortedMap` vs. `NavigableMap`.

**Comment by cpovirk** [comment]
RE: the code snippet:

`myMultimap.get("foo")` returns a `SortedSet<String>`. Are you thinking of a call like `myMultimap.get("foo").first()`?

To the more general point, I agree that this is an unfortunate interaction. You're right that we won't change `SortedSetMultimap` (on account of binary compatibility). That said, we could do _some_ work to reduce the number of places that we _return_ `SortedSet` -- including some methods on `TreeMultimap` (though there's no more to do for `get` specifically, as it already returns `NavigableSet`). Coincidentally, I had just recently filed an issue about this internally (154817068, for my reference).

Your proposal sounds good to me, for whatever that's worth :)

**Comment by Stephan202** [comment]
> `myMultimap.get("foo")` returns a `SortedSet<String>`. Are you thinking of a call like `myMultimap.get("foo").first()`?

That's what you get for modifying code for demo purposes :man_facepalming: . Indeed, I meant something like that :)

**Comment by cushon** [comment]
Per https://github.com/google/error-prone/issues/1646#issuecomment-756342637 we don't warn on this case anymore
